tldw_Server_API/tests/Config/test_config_providers_endpoints.py

```
@@ -0,0 +1,360 @@
+            response = await validate_provider_key(body)
+
+        assert response.valid is True
+        # Verify the URL contained the key as query param
+        call_args = mock_client.get.call_args
+        url_used = call_args[0][0] if call_args[0] else call_args[1].get("url", "")
+        assert "key=google-test-key" in url_used
+
+    @pytest.mark.asyncio
+    async def test_timeout_handled_gracefully(self):
+        """Timeouts should return a clear error."""
+        async def _slow_validate(*args, **kwargs):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_slow_validate`

(ANN202)